onnxruntime/core/providers/qnn/builder/opbuilder/conv_op_builder.cc

```diff
@@ -78,154 +107,392 @@ Status ConvOpBuilder::IsOpSupported(QnnModelWrapper& qnn_model_wrapper,
-                                                                       do_op_validation,
-                                                                       is_graph_input));
-        } else {
-          ORT_THROW("Unexpected operator %s", node_unit.OpType());
-        }
-        input_tensor_name = transpose_output_name;
+
+      std::vector<uint32_t> shape = {
+          input0_info.shape[0],  // N
+          1,                     // Height == 1
+          input0_info.shape[1],  // Width
+          input0_info.shape[2]   // Channels
+      };
+
+      if (!input0_info.is_initializer) {
+        // Add Reshape node to transform 1D input to 2D (i.e., set height to 1).
+        // We don't need to do this for initializers, because the number of elements does not change. We can just
+        // modify the shape dimensions.
+        bool is_graph_input = qnn_model_wrapper.IsGraphInput(input0_name);
+        ORT_RETURN_IF_ERROR(qnn_model_wrapper.AddReshapeNode(input0_name,
+                                                             conv_input0_name,
+                                                             input0_info.shape,
+                                                             shape,
+                                                             input0_info.qnn_data_type,
+                                                             input0_info.quant_param,
+                                                             do_op_validation,
+                                                             is_graph_input));
       }
-      input_shape = new_input_shape;
+
+      Qnn_TensorType_t tensor_type = GetInputTensorType(qnn_model_wrapper, conv_input0_name);
+      QnnTensorWrapper input_tensorwrapper(conv_input0_name, tensor_type, input0_info.qnn_data_type, input0_info.quant_param,
+                                           std::move(shape), std::move(unpacked_tensor));
+      ORT_RETURN_IF_NOT(qnn_model_wrapper.AddTensorWrapper(std::move(input_tensorwrapper)), "Failed to add tensor.");
+    } else {
+      LOGS(logger, VERBOSE) << "Tensor already added, skip it: " << input0_name;
     }
-    input_names.push_back(input_tensor_name);
+  }
 
-    if (qnn_model_wrapper.IsQnnTensorWrapperExist(input_tensor_name)) {
-      LOGS(logger, VERBOSE) << "Tensor already added, skip it: " << input_tensor_name;
-      continue;
+  //
+  // Input 1: weight
+  // We need to first reshape the weight inorder to handle 1D convolutions with the Conv2d operator.
+  // Next, we have to transpose the weight because ORT layout transformations do not change the weight layout.
+  //
+  {
+    const std::string& input1_name = inputs[1].node_arg.Name();
+    OnnxInputInfo input_info = {};
+    ORT_RETURN_IF_ERROR(qnn_model_wrapper.GetOnnxInputInfo(inputs[1], is_quantized_model, input_info));
+
+    std::string conv_weight_input_name = input_info.is_initializer ? input1_name : input1_name + "_trans_qnn_ep";
```

Comment:
is there some naming convention followed? noticed that for the 2d case it's _trans and here it's _trans_qnn_ep

Reply:
I'm not sure if there's been a convention, but perhaps we can do something like "_ort_qnn_ep_<inserted_op>"?

Reply:
that's fine. i was just wondering if we were being consistent across qnn ep when we add these.